src/veritysetup/veritysetup.c

```diff
@@ -107,7 +130,92 @@ static int parse_options(const char *options) {
                 else if (streq(word, "panic-on-corruption"))
                         arg_activate_flags |= CRYPT_ACTIVATE_PANIC_ON_CORRUPTION;
 #endif
-                else if ((val = startswith(word, "root-hash-signature="))) {
+                else if ((val = startswith(word, "no-superblock="))) {
+
+                        r = parse_boolean(val);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+
+                        arg_no_superblock = r;
+                } else if ((val = startswith(word, "format="))) {
+
+                        r = safe_atoi(val, &arg_format);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                } else if ((val = startswith(word, "data-block-size="))) {
+                        uint64_t sz;
+
+                        r = safe_atou64(val, &sz);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                        if (!block_size_is_valid(sz))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse %s: %m", word);
+
+                        arg_data_block_size = sz;
+                } else if ((val = startswith(word, "hash-block-size="))) {
+                        uint64_t sz;
+
+                        r = parse_size(val, 1024, &sz);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                        if (!block_size_is_valid(sz))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse %s: %m", word);
+
+                        arg_hash_block_size = sz;
+                } else if ((val = startswith(word, "data-blocks="))) {
+
+                        r = safe_atou64(val, &arg_data_blocks);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                } else if ((val = startswith(word, "hash-offset="))) {
+
+                        r = safe_atou64(val, &arg_hash_offset);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
+                } else if ((val = startswith(word, "salt="))) {
+
+                        if (isempty(val)) {
+                                arg_salt = mfree(arg_salt);
+                                arg_salt_size = 32;
+                        } else if (streq(val, "-")) {
+                                arg_salt = mfree(arg_salt);
+                                arg_salt_size = 0;
+                        } else {
+                                r = unhexmem(val, strlen(val), &arg_salt, &arg_salt_size);
```

Comment:
Compiler's unhappy:

```
../src/veritysetup/veritysetup.c: In function ‘parse_options’:
../src/veritysetup/veritysetup.c:184:75: error: passing argument 4 of ‘unhexmem’ from incompatible pointer type [-Werror=incompatible-pointer-types]
                                 r = unhexmem(val, strlen(val), &arg_salt, &arg_salt_size);
                                                                           ^
In file included from ../src/veritysetup/veritysetup.c:10:0:
../src/basic/hexdecoct.h:22:19: note: expected ‘size_t * {aka unsigned int *}’ but argument is of type ‘uint64_t * {aka long long unsigned int *}’
 static inline int unhexmem(const char *p, size_t l, void **mem, size_t *len) {
```

Reply:
I will fix this.

Reply:
Waiting for the result of the CI and see if the warning is raised; I have not looked at it to be honest.

Reply:
```
../src/veritysetup/veritysetup.c: In function â€˜parse_optionsâ€™:
../src/veritysetup/veritysetup.c:201:75: error: passing argument 4 of â€˜unhexmemâ€™ from incompatible pointer type [-Werror=incompatible-pointer-types]
  201 |                                 r = unhexmem(val, strlen(val), &arg_salt, &arg_salt_size);
      |                                                                           ^~~~~~~~~~~~~~
      |                                                                           |
      |                                                                           uint64_t * {aka long long unsigned int *}
In file included from ../src/veritysetup/veritysetup.c:10:
../src/basic/hexdecoct.h:22:73: note: expected â€˜size_t *â€™ {aka â€˜unsigned int *â€™} but argument is of type â€˜uint64_t *â€™ {aka â€˜long long unsigned int *â€™}
   22 | static inline int unhexmem(const char *p, size_t l, void **mem, size_t *len) {
      |                                                                 ~~~~~~~~^~~
cc1: all warnings being treated as errors
```

Fixing...